### PR TITLE
New Component Concepts and Improvements

### PR DESCRIPTION
**Pull request by bvkimball**
**Description**

TLDR;
feat(Avatar): New Avatar Component
feat(BreadCrumb): New Breadcrumb Component
feat(Aside): New Aside Component
feat(ColorPicker): New Color Picker Component
feat(Field): New Field Layout Component
feat(Progress): New Progress Bar Component
fix(Radio): Radio button groups more reactive now
fix(Overlay): Works with multiple overlays now
feat(Button): added proper `size` styling
feat(DatePicker): Added Multi Date selection
feat(DatePicker): Added Range Picker Input

#### **Verify that...**

- [x] Any related demos were added and `npm start` and `npm run build` still works
- [ ] New demos work in `Safari`, `Chrome` and `Firefox`
- [ ] `npm run lint` passes
- [ ] `npm test` passes and code coverage is increased
- [ ] `npm run build` still works

#### **Bullhorn Internal Developers**
- [ ] Run `Novo Automation`
- [ ] Run `BBO Automation`

##### **Screenshots**

### CODE REVIEW

File: projects/novo-elements/src/elements/avatar/Avatar.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        const svgHtml: any = window.btoa(unescape(encodeURIComponent(top.innerHTML)));
+        src = `data:image/svg+xml;base64, ${svgHtml}`;
+      }
+      this.src = this.sanitizer.bypassSecurityTrustUrl(src);
+    }
+  }
+
+  setPrefixedValue(elm: any, prop: any, value: any): any {
+    const prefixes: any = ['-moz-', '-webkit-', '-o-', '-ms-', '-khtml-'];
```
Comment (tbo47) [inline projects/novo-elements/src/elements/avatar/Avatar.ts:146]: Can we not use `any` as much as possible?

File: projects/novo-elements/src/elements/avatar/Avatar.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      elm.style[prop] = value;
+      if (elm.style[prop] !== starting) {
+        return;
+      }
+    } catch (e) {
+      // no op
+    }
+
+    // Try prefixes
+    for (i = 0; i < prefixes.length; ++i) {
```
Comment (tbo47) [inline projects/novo-elements/src/elements/avatar/Avatar.ts:165]: Can `i` and `v` be `const` and defined in the `for` loop?

File: projects/novo-elements/src/elements/avatar/Avatar.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
+:host {
+  img {
+    width: 30px;
+    height: 30px;
+    border-radius: 2em;
+  }
+  // @each $basic, $color in $all-colors {
```
Comment (ndickerson) [inline projects/novo-elements/src/elements/avatar/Avatar.scss:7]: I'm not a fan of commented out code. If there's a reason we want code to be commented out, then please add a comment at the top stating why the commented out code is there.

File: projects/novo-elements/src/elements/field/field.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+@Component({
+  selector: 'novo-field',
+  templateUrl: './field.html',
+  styleUrls: ['./field.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+    class: 'novo-field',
+    '[class.novo-field-appearance-horizontal]': 'appearance=="horizontal"',
+    '[class.novo-field-appearance-vertical]': 'appearance=="vertical"',
+    // '[class.novo-field-appearance-standard]': 'appearance == "standard"',
```
Comment (ndickerson) [inline projects/novo-elements/src/elements/field/field.ts:40]: More commented out code

**Comment by ndickerson** [comment]
There are some neat new components here. There are also some major upgrades to existing components. Some thoughts:

1. When we are closer to getting these changes merged, we will want to run all Novo UI Automation nightly against the Novo feature branch in order to assure full coverage of the changes, like we did after major overlay improvements.
2. With major Date Picker modifications, we will want to ensure we test date pickers in Novo across all supported languages, as there have been several bug fixes in the past around our internationalized date pickers. 
3. We may want to consider upping the major release version if enough of the contract for existing components has changed

File: projects/novo-elements/src/elements/avatar/Avatar.scss (hunk 3 above)
Comment (bvkimball) [reply projects/novo-elements/src/elements/avatar/Avatar.scss:7]: copy pasta from the original component.

File: projects/novo-elements/src/elements/field/field.ts (hunk 4 above)
Comment (bvkimball) [reply projects/novo-elements/src/elements/field/field.ts:40]: still in dev

**Comment by bvkimball** [comment]
moved to f/golden

**Pull request closed without merging**